### PR TITLE
[python3] Install pkg-config files on Windows platforms

### PR DESCRIPTION
**Pull request by mkhon**
Install pkg-config configuration files on Windows platforms (msbuild builds)

- #### What does your PR fix?  

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?  
all

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?  
Yes

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?  

**If you are still working on the PR, open it as a Draft: https://github.blog/2019-02-14-introducing-draft-pull-requests/**

### CODE REVIEW

File: ports/python3/portfile.cmake [hunk 1]
```diff
@@ -34,6 +34,7 @@ if(VCPKG_TARGET_IS_WINDOWS OR VCPKG_TARGET_IS_UWP)
     if("${WINSDK_VERSION}" VERSION_GREATER_EQUAL "10.0.22000")
         list(APPEND PATCHES "0007-workaround-windows-11-sdk-rc-compiler-error.patch")
     endif()
+    list(APPEND PATCHES "0008-python-embed.pc.patch")
```
Comment (Hoikas) [inline ports/python3/portfile.cmake:37]: There is already a patch with the number 0008.

File: ports/python3/portfile.cmake [hunk 2]
```diff
@@ -158,6 +159,25 @@ if(VCPKG_TARGET_IS_WINDOWS OR VCPKG_TARGET_IS_UWP)
...
+        configure_file("${SOURCE_PATH}/Misc/python.pc.in" "${CURRENT_PACKAGES_DIR}/lib/pkgconfig/python-${VERSION}.pc" @ONLY)
+        configure_file("${SOURCE_PATH}/Misc/python-embed.pc.in" "${CURRENT_PACKAGES_DIR}/lib/pkgconfig/python-${VERSION}-embed.pc" @ONLY)
+
+        set(exec_prefix "\${prefix}/../tools/${PORT}")
+        set(includedir [[${prefix}/../include]])
+        set(ABISUFFIX "_d")
+        configure_file("${SOURCE_PATH}/Misc/python.pc.in" "${CURRENT_PACKAGES_DIR}/debug/lib/pkgconfig/python-${VERSION}.pc" @ONLY)
+        configure_file("${SOURCE_PATH}/Misc/python-embed.pc.in" "${CURRENT_PACKAGES_DIR}/debug/lib/pkgconfig/python-${VERSION}-embed.pc" @ONLY)
+    vcpkg_restore_env_variables(VARS prefix exec_prefix libdir includedir VERSION)
+    vcpkg_fixup_pkgconfig()
```
Comment (Hoikas) [inline ports/python3/portfile.cmake:179]: Did you intend to only do this for Windows?

File: ports/python3/portfile.cmake [hunk 3]
```diff
@@ -158,6 +159,25 @@ if(VCPKG_TARGET_IS_WINDOWS OR VCPKG_TARGET_IS_UWP)
...
+        set(includedir [[${prefix}/include]])
+        set(ABIVERSION "${PYTHON_VERSION_MAJOR}${PYTHON_VERSION_MINOR}")
+        set(VERSION "${PYTHON_VERSION_MAJOR}.${PYTHON_VERSION_MINOR}")
+
+        configure_file("${SOURCE_PATH}/Misc/python.pc.in" "${CURRENT_PACKAGES_DIR}/lib/pkgconfig/python-${VERSION}.pc" @ONLY)
+        configure_file("${SOURCE_PATH}/Misc/python-embed.pc.in" "${CURRENT_PACKAGES_DIR}/lib/pkgconfig/python-${VERSION}-embed.pc" @ONLY)
+
+        set(exec_prefix "\${prefix}/../tools/${PORT}")
+        set(includedir [[${prefix}/../include]])
+        set(ABISUFFIX "_d")
```
Comment (Hoikas) [inline ports/python3/portfile.cmake:175]: Why not append this to the abiflags so the patch is not needed?

**Review by Hoikas — COMMENTED** [review]
These changes assume that both release and debug builds are being built. It needs to be made safe for release only and debug only triplets.

**Comment by LilyWangLL** [comment]
Thanks for your PR. Please run command `vcpkg x-add-version --all` then commit the changes.

File: ports/python3/portfile.cmake (hunk 3 above)
Comment (mkhon) [reply ports/python3/portfile.cmake:175]: ABIFLAGS also alters Cflags (includedir) so we can not do it using ABIFLAGS unfortunately

File: ports/python3/portfile.cmake (hunk 2 above)
Comment (mkhon) [reply ports/python3/portfile.cmake:179]: I think pkg-config files are needed on all architectures

File: ports/python3/portfile.cmake (hunk 1 above)
Comment (mkhon) [reply ports/python3/portfile.cmake]: Fixed

**Comment by mkhon** [comment]
> These changes assume that both release and debug builds are being built. It needs to be made safe for release only and debug only triplets.

Fixed now

File: ports/python3/portfile.cmake (hunk 2 above)
Comment (mkhon) [reply ports/python3/portfile.cmake]: I realized that the patch 0009-python-embed.pc.patch is needed on all arches. Fixed now

**Review by LilyWangLL — APPROVED** [review]

File: ports/python3/portfile.cmake (hunk 2 above)
Comment (mkhon) [reply ports/python3/portfile.cmake]: Yes, thos whole if() block is for Windows (msbuild)

File: ports/python3/portfile.cmake (hunk 3 above)
Comment (mkhon) [reply ports/python3/portfile.cmake]: Looks like there is a related problem with ABIFLAGS in Linux build - I will submit a separate PR about this (requires more work and not related to this one)

File: ports/python3/portfile.cmake [hunk 4]
```diff
@@ -158,6 +159,30 @@ if(VCPKG_TARGET_IS_WINDOWS OR VCPKG_TARGET_IS_UWP)
         )
     endif()
 
+    vcpkg_backup_env_variables(VARS prefix exec_prefix libdir includedir VERSION)
```
Comment (BillyONeal) [inline ports/python3/portfile.cmake:162]: Maybe I'm confused by my understanding is that `configure_file` only looks at CMake variables, not environment variables, so this call to `vcpkg_backup_env_variables` and `vcpkg_restore_env_variables` doesn't seem to have an effect?

Comment (JackBoosY) [reply ports/python3/portfile.cmake:162]: Ping @mkhon for response.

Comment (mkhon) [reply ports/python3/portfile.cmake]: Yes, @BillyONeal is right - I updated the PR

**Comment by BillyONeal** [comment]
Thanks!

**Pull request merged**